In [2]:
import pandas as pd

In [63]:
data = pd.read_csv('New1/data/adult.data.csv')
data.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

#### 1. Сколько мужчин и женщин (признак sex) представлено в этом наборе данных?

In [4]:
data['sex'].value_counts()

Male      21790
Female    10771
Name: sex, dtype: int64

#### Ответ: В этом наборе данных представлено 21790 мужчин и 10771 женщина

#### 2. Каков средний возраст (признак age) женщин?

In [6]:
data[(data['sex'] == 'Female')]['age'].mean()

36.85823043357163

#### Средний возраст женщины 36 лет

#### 3. Какова доля граждан Германии (признак native-country)?

In [14]:
data['native-country'].value_counts(normalize=True) 

United-States                 0.895857
Mexico                        0.019748
?                             0.017905
Philippines                   0.006081
Germany                       0.004207
Canada                        0.003716
Puerto-Rico                   0.003501
El-Salvador                   0.003255
India                         0.003071
Cuba                          0.002918
England                       0.002764
Jamaica                       0.002488
South                         0.002457
China                         0.002303
Italy                         0.002242
Dominican-Republic            0.002150
Vietnam                       0.002058
Guatemala                     0.001966
Japan                         0.001904
Poland                        0.001843
Columbia                      0.001812
Taiwan                        0.001566
Haiti                         0.001351
Iran                          0.001321
Portugal                      0.001136
Nicaragua                

#### Процент граждан Германии составляет 0.42%

#### 4-5. Каковы средние значения и среднеквадратичные отклонения возраста тех, кто получает более 50K в год (признак salary) и тех, кто получает менее 50K в год?

In [24]:
data.groupby('salary')['age'].describe()

count       mean        std   min   25%   50%   75%   max
salary                                                             
<=50K   24720.0  36.783738  14.020088  17.0  25.0  34.0  46.0  90.0
>50K     7841.0  44.249841  10.519028  19.0  36.0  44.0  51.0  90.0

#### Среднее знач, стд: меньше 50К - 36, 14, больше 50К - 44, 10.   

#### 6. Правда ли, что люди, которые получают больше 50k, имеют как минимум высшее образование? (признак education – Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters или Doctorate)

In [29]:
more_50k = data[(data['salary'] == '>50K')]
more_50k['education'].value_counts()

Bachelors       2221
HS-grad         1675
Some-college    1387
Masters          959
Prof-school      423
Assoc-voc        361
Doctorate        306
Assoc-acdm       265
10th              62
11th              60
7th-8th           40
12th              33
9th               27
5th-6th           16
1st-4th            6
Name: education, dtype: int64

#### Не правда

#### 7. Выведите статистику возраста для каждой расы (признак race) и каждого пола. Используйте groupby и describe. Найдите таким образом максимальный возраст мужчин расы Amer-Indian-Eskimo.

In [32]:
data.groupby(['race'])['age'].describe()

count       mean        std   min   25%   50%   75%  \
race                                                                        
Amer-Indian-Eskimo    311.0  37.173633  12.447130  17.0  28.0  35.0  45.5   
Asian-Pac-Islander   1039.0  37.746872  12.825133  17.0  28.0  36.0  45.0   
Black                3124.0  37.767926  12.759290  17.0  28.0  36.0  46.0   
Other                 271.0  33.457565  11.538865  17.0  25.0  31.0  41.0   
White               27816.0  38.769881  13.782306  17.0  28.0  37.0  48.0   

                     max  
race                      
Amer-Indian-Eskimo  82.0  
Asian-Pac-Islander  90.0  
Black               90.0  
Other               77.0  
White               90.0

#### Max age for Amber-Indian-Eskimo - 82 ages

#### 8. Среди кого больше доля зарабатывающих много (>50K): среди женатых или холостых мужчин (признак marital-status)? Женатыми считаем тех, у кого marital-status начинается с Married (Married-civ-spouse, Married-spouse-absent или Married-AF-spouse), остальных считаем холостыми.

In [45]:
data['marital-status'] = data['marital-status'].str.startswith('Married')

AttributeError: Can only use .str accessor with string values, which use np.object_ dtype in pandas

In [44]:
pd.crosstab(data['marital-status'], data['salary'])

salary          <=50K  >50K
marital-status             
False           16039  1105
True             8681  6736

#### Среди тех кто зарабатывает много больше женатых

#### 9. Какое максимальное число часов человек работает в неделю (признак hours-per-week)? Сколько людей работают такое количество часов и каков среди них процент зарабатывающих много?

In [46]:
data['hours-per-week'].describe()

count    32561.000000
mean        40.437456
std         12.347429
min          1.000000
25%         40.000000
50%         40.000000
75%         45.000000
max         99.000000
Name: hours-per-week, dtype: float64

#### Максимальное число часов - 99 ч.

In [56]:
work_99_hour = data[(data['hours-per-week'] == 99)]
work_99_hour['hours-per-week'].count()

85

### 85 человек из выборки работают 99 часов в неделю

In [58]:
pd.crosstab(work_99_hour['hours-per-week'], work_99_hour['salary'], normalize=True)

salary             <=50K      >50K
hours-per-week                    
99              0.705882  0.294118

#### Процент зарабатывающих много среди этих людей равняется 29.4%

#### 10. Посчитайте среднее время работы (hours-per-week) зарабатывающих мало и много (salary) для каждой страны (native-country).

In [62]:
countries_salary = data.groupby(['native-country', 'salary'])['hours-per-week'].mean()
countries_salary['Japan']

salary
<=50K    41.000000
>50K     47.958333
Name: hours-per-week, dtype: float64